plugins/gitlab/src/class.ts

```diff
@@ -206,17 +209,23 @@ export class GitlabZoneApi extends GitlabApi {
     })
   }
 
-  public async getOrCreateInfraProject(zone: string) {
+  public async getOrCreateInfraProject(zone: string): Promise<ProjectSchema> {
+    return this.infraProjectsByZoneSlug.get(zone) ?? await this.createInfraProject(zone)
+  }
+
+  private async createInfraProject(zone: string): Promise<ProjectSchema> {
     const infraGroup = await this.getOrCreateInfraGroup()
     // Get or create projects_root_dir/infra/zone
     const infraProjects = await this.api.Groups.allProjects(infraGroup.id)
```

Comment:
Tu penses qu'on peut aussi optimiser la request avc les `AllGroupProjectsOptions` [this](https://github.com/jdalrymple/gitbeaker/blob/40.6.0/packages/core/src/resources/Groups.ts#L93), no ?